setuptools/command/easy_install.py

```diff
@@ -2015,7 +2020,13 @@ def is_python_script(script_text, filename):
     from os import chmod as _chmod
 except ImportError:
```

Comment:
It's up to the checker's implementation. mypy will use the first definition, pyright will use the second.
Currently, mypy incorrectly thinks that you can pass in kwargs (so at least adding `**kwargs` to the Jython compatibility version would be an improvement)
and pyright would completely loose type safety.
This only affects setuptools internally though, so it's a "type safety risk" (a very minor one) you can take.

Pylance:
![image](https://github.com/pypa/setuptools/assets/1350584/045fa9c8-13e4-4f2b-888f-d7d886614739)

pyright:
`setuptools\command\easy_install.py:2030:21 - information: Type of "_chmod" is "(*args: object, **kwargs: object) -> None"`

mypy:
`setuptools\command\easy_install.py:2032: note: Revealed type is "def (path: Union[builtins.int, builtins.str, builtins.bytes, os.PathLike[builtins.str], os.PathLike[builtins.bytes]], mode: builtins.int, *, dir_fd: Union[builtins.int, None] =, follow_symlinks: builtins.bool =)"`

Reply:
I think it is worth doing something like the following, so that we remove the need for also defining `_FileDescriptorOrPath`

```python
try:
    from os import chmod as _chmod
except ImportError:  # type: ignore
    def _chmod(*args):
        pass

Reply:
A `type: ignore` comment at that position doesn't do anything. But in my last update I did just suppress this type error. Is it satisfactory?

> Then we can also remove the `_FileDescriptorOrPath` definition, right?

Yep.

Reply:
I could also do something a bit cheeky: just leave this untyped for this PR. So mypy won't raise an issue (because I have it currently configured to not check untyped functions). Deferring these changes to #4192

Reply:
I think the way you sorted it out should just be fine...
Thank you very much @Avasam.